flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/RestHandlerConfiguration.java

```diff
@@ -96,13 +105,20 @@ public static RestHandlerConfiguration fromConfiguration(Configuration configura
 
         final boolean webSubmitEnabled = configuration.getBoolean(WebOptions.SUBMIT_ENABLE);
         final boolean webCancelEnabled = configuration.getBoolean(WebOptions.CANCEL_ENABLE);
+        final boolean webRescaleSupported =
+                JobManagerOptions.SchedulerType.Adaptive.equals(
+                                configuration.get(JobManagerOptions.SCHEDULER))
+                        && configuration.get(JobManagerOptions.SCHEDULER_MODE) == null;
+        final boolean webRescaleEnabled =
```

Comment:
Do we want this option to be purely for the UI or also for the REST API as a whole? If the latter, then we could also disable the requirement handlers directly in the endpoint.

Reply:
I was looking at the job cancellation (webCancelEnabled), and that doesn't seem to be the case for it. If this is correct, should we leave it as is and create a follow to implement this for all "web UI mutability features"?

Reply:
`web.submit.enable` can also disable the jar handlers :sweat_smile: 

I don't really like having 1 option for both the UI and REST API.

If we want to allow users to disable this, then I'd prefer a separate rest option (that would then also potentially disable the UI option).
Not sure if we need that though in the first place.

Reply:
For now, I'd leave it as is (just a UI option) and develop a systematic solution to align this across all three options (submit, cancel, rescale). Otherwise, it will create more confusion.

https://issues.apache.org/jira/browse/FLINK-31734